examples/evm-price-feed/src/tally_phase.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::Result;
+use ethabi::{Token, ethereum_types::U256};
+use seda_sdk_rs::{Process, elog, get_reveals, log};
+
+/// Executes the tally phase within the SEDA network.
+/// This phase aggregates the results (e.g., price data) revealed during the execution phase,
+/// calculates the median value, and submits it as the final result.
+/// Note: The number of reveals depends on the replication factor set in the data request parameters.
+pub fn tally_phase() -> Result<()> {
+    // Retrieve consensus reveals from the tally phase.
+    let reveals = get_reveals()?;
+    let mut revealed_prices: Vec<Vec<u128>> = Vec::with_capacity(reveals.len());
+
+    // Iterate over each reveal, parse its content as an unsigned integer (u128), and store it in the prices array.
+    for reveal in reveals {
+        let prices = match serde_json::from_slice::<Vec<u128>>(&reveal.body.reveal) {
+            Ok(prices) => prices,
+            Err(err) => {
+                elog!("Failed to parse revealed prices: {err}");
+                continue;
+            }
+        };
+
+        revealed_prices.push(prices);
+    }
+
+    if revealed_prices.is_empty() {
+        // If no valid prices were revealed, report an error indicating no consensus.
+        Process::error("No consensus among revealed results".as_bytes());
+        return Ok(());
+    }
+
+    // If there are valid prices revealed, calculate the median price from price reports.
+    let final_prices = median_each_asset(&revealed_prices)?;
+    log!("Final median prices: {:?}", final_prices);
+
+    let encoded_result = ethabi::encode(&[Token::Array(
+        final_prices
+            .into_iter()
+            .map(|p| Token::Uint(U256::from(p)))
+            .collect(),
+    )]);
```

Comment:
IMO preferable to have `median_each_asset` return a final `Token::Array` as to avoid having to iterate the whole list again. It still logs fine.